File: backend/app/schemas/asset.py
"""Asset schemas."""

from datetime import datetime
from pydantic import BaseModel, ConfigDict


class AssetBase(BaseModel):
    """Base asset schema."""
    name: str
    asset_type: str
    serial_number: str | None = None
    location: str | None = None
    description: str | None = None


class AssetCreate(AssetBase):
    """Schema for creating a new asset."""
    pass


class AssetUpdate(BaseModel):
    """Schema for updating an existing asset."""
    name: str | None = None
    asset_type: str | None = None
    serial_number: str | None = None
    location: str | None = None
    description: str | None = None


class AssetResponse(AssetBase):
    """Schema for asset response."""
    id: int
    created_by_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend-react/src/types/asset.ts
// 资产相关类型，对应后端 schemas/asset.py。
// 资产（Asset）代表一台实体设备，如交换机、路由器，可关联多张工单。

export interface Asset {
  id: number
  name: string
  asset_type: string
  serial_number: string | null
  location: string | null
  description: string | null
  created_by_id: number
  created_at: string
  updated_at: string
}

export interface AssetCreate {
  name: string
  asset_type: string
  serial_number?: string
  location?: string
  description?: string
}
